src/bt_peer_connection.cpp

```diff
@@ -1618,7 +1617,7 @@ namespace libtorrent
 	{
 		char buf[35];
 		char* ptr = buf + 6;
-		detail::write_uint8(type, ptr);
+		detail::write_uint8(std::uint8_t(type), ptr);
```

Comment:
it seems strange to say the type twice. in a sense, since the type passed to the function is explicit already, it seems like this cast should be implied. The only way I can think of doing that though would be to make ``write_uint8()`` a template